### PR TITLE
SAA-588: Added attendance created and amended entity listener, events and feature switches

### PR DESCRIPTION
**Pull request by timharrison-moj**
Events are configured as OFF for the first deployment - will switch on Monday.

Changes:
* Add entity listeners on attendance create or update.
* Add events for create and update attendance.
* Add feature switches for new events.
* Updated the name of env vars and events to match agreed confluence page with Syscon - the allocation amended event had 4 parts to it, instead of 3. i.e. `activities.prisoner.allocation.amended` when should be `activities.prisoner.allocation-amended` - with a hyphen, though the enum and env var still uses the underscore (and still matches using Spring configuration rules).

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/EventsService.kt [hunk 1]
```diff
@@ -63,14 +67,30 @@ enum class OutboundEvent(val eventType: String) {
...
-  PRISONER_ALLOCATION_AMENDED("activities.prisoner.allocation.amended") {
+  PRISONER_ALLOCATION_AMENDED("activities.prisoner.allocation-amended") {
     override fun event(additionalInformation: AdditionalInformation) =
       OutboundHMPPSDomainEvent(
         eventType = eventType,
         additionalInformation = additionalInformation,
         description = "A prisoner allocation has been amended in the activities management service",
       )
   },
+  PRISONER_ATTENDANCE_CREATED("activities.prisoner.attendance-created") {
```
[the remark this answers is not in the document]
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/EventsService.kt:78]: Yes, good spot. I'll add a couple to the attendance integration test now.

[the remark this answers is not in the document]
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/EventsService.kt:78]: Added these integration test - good job too - as it highlighted a couple of issues with these. Now sorted.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/Attendance.kt [hunk 2]
```diff
@@ -22,10 +23,11 @@ import uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.model.Attendanc
 
 @Entity
 @Table(name = "attendance")
+@EntityListeners(AttendanceEntityListener::class)
 data class Attendance(
   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
-  val attendanceId: Long = -1,
+  val attendanceId: Long = 0,
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/Attendance.kt:31]: This was key ... with -1 in the Id, the entity listeners were seeing 1) A created event, followed by 2) An amended event. With 0 here, we just see the created event, so 0 or null look like the preferred default values for generated ID fields.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/EventsService.kt [hunk 3]
```diff
@@ -63,14 +67,30 @@ enum class OutboundEvent(val eventType: String) {
         description = "A prisoner has been allocated to an activity in the activities management service",
       )
   },
-  PRISONER_ALLOCATION_AMENDED("activities.prisoner.allocation.amended") {
+  PRISONER_ALLOCATION_AMENDED("activities.prisoner.allocation-amended") {
     override fun event(additionalInformation: AdditionalInformation) =
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/EventsService.kt:71]: Event types should be 3-part, not 4-part strings, and match the confluence page for sync events.

**Review by matt-crowson-moj — APPROVED** [review]
LGTM.  We will have to make the same change for ID's across all entities to be safe.

**Review by matt-crowson-moj — APPROVED** [review]

**Pull request merged**